bridge/bridge.go

```diff
@@ -394,6 +402,7 @@ func (b *Bridge) remove(containerId string, deregister bool) {
 		b.deadContainers[containerId] = &DeadContainer{b.config.RefreshTtl, b.services[containerId]}
 	}
 	delete(b.services, containerId)
+	b.markContainerAsDying(containerId)
```

Comment:
minor, but imo we should markContainerAsDying first and then delete it from `b,services`. Just to avoid (potentially?) re-adding it right away, before we marked it as dying

Reply:
This case will not happen, because the Bridge will be unlocked only after we marked the container as dying. See here https://github.com/auto1-oss/registrator/blob/f3c55eedbdccb7ff35db62861b445dfc49c213de/bridge/bridge.go#L382

Reply:
Bridge in not locked when we try to add the container. How do you avoid the race condition?
So from my pespective, we technically can run into scneario where in between
`delete(b.services, containerId)` and `b.markContainerAsDying(containerId)` we call `add`. Mean, contaienr was alerady delete from services, but not yet added into `dyingContainers`. There is not synchronization in `add` method.

Reply:
It looks like all `add` functions invocation are guarded with `Muttex`. We should bw find here.